modules/redshift-loader/src/main/scala/com/snowplowanalytics/snowplow/loader/redshift/Redshift.scala

```diff
@@ -58,26 +50,45 @@ object Redshift {
 
           override val requiresEventsColumns: Boolean = false
 
-          override def updateTable(migration: Migration): Block = {
-            val ddlFile = MigrationGenerator.generateMigration(migration, 4096, Some(schema))
-
-            val (preTransaction, inTransaction) = ddlFile.statements.foldLeft((NoPreStatements, NoStatements)) {
-              case ((preTransaction, inTransaction), statement) =>
-                statement match {
-                  case s @ AlterTable(_, _: AlterType) =>
-                    (Item.AlterColumn(Fragment.const0(s.toDdl)) :: preTransaction, inTransaction)
-                  case s @ AlterTable(_, _) =>
-                    (preTransaction, Item.AddColumn(Fragment.const0(s.toDdl), ddlFile.warnings) :: inTransaction)
-                  case _ => // We explicitly support only ALTER TABLE here; also drops BEGIN/END
-                    (preTransaction, inTransaction)
-                }
-            }
-
-            val target = SchemaKey(migration.vendor, migration.name, "jsonschema", migration.to)
-            Block(preTransaction.reverse, inTransaction.reverse, Entity.Table(schema, target))
+          override def updateTable(shredModel: ShredModel.GoodModel, currentSchemaKey: SchemaKey): Block = {
+            val outTransactions = shredModel.migrations.outTransaction(Some(currentSchemaKey))
+            val inTransactions = shredModel.migrations.inTransaction(Some(currentSchemaKey))
+            val outTransactionToSql =
+              outTransactions.map { varcharExtension =>
+                s"""  ALTER TABLE $schema.${shredModel.tableName}
+                 |    ALTER COLUMN "${varcharExtension.old.columnName}" TYPE ${varcharExtension.newEntry.columnType.show};
+                 |""".stripMargin
+              }.mkString
+            val inTransactionToSql =
+              inTransactions.map { columnAddition =>
+                s"""  ALTER TABLE $schema.${shredModel.tableName}
+                   |    ADD COLUMN "${columnAddition.column.columnName}" ${columnAddition.column.columnType.show} ${columnAddition.column.compressionEncoding.show};
+                   |""".stripMargin
+              } match {
+                case Nil => s"""|
+                                |-- NO ADDED COLUMNS CAN BE EXPRESSED IN SQL MIGRATION
+                                |
+                                |COMMENT ON TABLE $schema.${shredModel.tableName} IS '${shredModel.schemaKey.toSchemaUri}';
+                                |""".stripMargin
+                case h :: t => s"""|
+                                  |BEGIN TRANSACTION;
```

Comment:
The doobie transactor automatically wraps our transactions with a `BEGIN` and `END`.  So you don't need to explicitly add them to statements.